libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentParser.java

```diff
@@ -135,6 +139,10 @@ public boolean isValue() {
 
     Map<String, String> mapStringsOrdered() throws IOException;
 
+    <T> Map<String, T> genericMap(MapValueParser<T> mapValueParser) throws IOException;
```

Comment:
we should add javadocs if we add new methods please

Reply:
Done.